### ISSUE TITLE
[3.2] UWP build fails after Thread modernization

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#46314]
**Godot version:**
`3.2` (2735a5498e01cbeb882b5adbc0e9fcfcd847df8f)
Might also affect `master` though UWP build is likely completely broken there currently for other reasons.

**OS/device including version:**
Fedora 32, cross-compiling for UWP with Wine and VS 2017.
(But likely same issue on regular Windows with VS 2019.)

**Issue description:**
Building for UWP fails with:
```
Z:\root\godot>call scons -j64 verbose=yes warnings=no progress=no platform=uwp production=yes tools=no target=release LINK="\"C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\HostX64\x86\link.exe\"" 
scons: Reading SConscript files ...

scons: warning: No version of Visual Studio compiler found - C/C++ compilers most likely not set correctly
File "Z:\root\godot\SConstruct", line 3, in <module>

scons: warning: No version of Visual Studio compiler found - C/C++ compilers most likely not set correctly
File "Z:\root\godot\SConstruct", line 64, in <module>
Compiled program architecture will be a x86 executable. (forcing bits=32).
Using custom build name: official
YASM is necessary for WebM SIMD optimizations.
WebM SIMD optimizations are disabled. Check if your CPU architecture, CPU bits or platform are supported!
Checking for C header file mntent.h... no
scons: done reading SConscript files.
scons: Building targets ...
scons: *** [platform\uwp\thread_uwp.uwp.opt.32.x86.obj] Source `platform\uwp\thread_uwp.cpp' not found, needed by target `platform\uwp\thread_uwp.uwp.opt.32.x86.obj'.
scons: building terminated because of errors.
```

**Steps to reproduce:**
- Set up build environment for UWP: https://docs.godotengine.org/en/stable/development/compiling/compiling_for_uwp.html
- `scons p=uwp`

**Minimal reproduction project:**
n/a

**Comment by akien-mga** [comment]
Tried a fix (see above commit) but it's not sufficient, some code needs to be fixed:
```
z:\root\godot\platform\uwp\app.h(81): error C2039: 'Thread': is not a member of 'Windows::System'
z:\root\godot\platform\uwp\app.h(81): error C2653: 'ing': is not a class or namespace name
z:\root\godot\platform\uwp\app.h(81): error C2143: syntax error: missing ';' before '^'
z:\root\godot\platform\uwp\app.h(81): error C2238: unexpected token(s) preceding ';'
z:\root\godot\platform\uwp\app.h(83): error C2039: 'Thread': is not a member of 'Windows::System'
z:\root\godot\platform\uwp\app.h(83): error C2653: 'ing': is not a class or namespace name
z:\root\godot\platform\uwp\app.h(83): error C2143: syntax error: missing ',' before '^'
```

**Comment by akien-mga** [comment]
Oops I pushed my non-working fix to the `3.2` branch by mistake as d1f023c35b8bed3fc839fdc5d587a839568aab44.
It's still a necessary change which exposes the errors from my previous comment though.

**Comment by akien-mga** [comment]
Fixed by d1f023c and #46322.
